mLRS/Common/sx-drivers/sx127x_driver.h

```diff
@@ -430,9 +430,153 @@ class Sx127xDriver : public Sx127xDriverCommon
+typedef enum {
+    SX2_IRQ_TX_DONE = SX1276_IRQ_TX_DONE,
+    SX2_IRQ_RX_DONE = SX1276_IRQ_RX_DONE,
+    SX2_IRQ_TIMEOUT = SX1276_IRQ_RX_TIMEOUT,
+    SX2_IRQ_ALL     = SX1276_IRQ_ALL,
+} SX2_IRQ_ENUM;
+
+
+class Sx127xDriver2 : public Sx127xDriverCommon
+{
+  public:
+
+    void SpiSelect(void) override
+    {
+        spib_select();
+        delay_ns(50); // datasheet says t1 = 25 ns, semtech driver doesn't do it, helps so do it
+    }
+
+    void SpiDeselect(void) override
+    {
+        delay_ns(50); // datasheet says t8 = 25 ns, semtech driver doesn't do it, helps so do it
+        spib_deselect();
+    }
+
+#ifndef DEVICE_HAS_DIVERSITY_SINGLE_SPI
+    void SpiTransfer(uint8_t* dataout, uint8_t* datain, uint8_t len) override
+    {
+        spib_transfer(dataout, datain, len);
+    }
+
+    void SpiRead(uint8_t* datain, uint8_t len) override
+    {
+        spib_read(datain, len);
+    }
+
+    void SpiWrite(uint8_t* dataout, uint8_t len) override
+    {
+        spib_write(dataout, len);
+    }
+#else
+    void SpiTransfer(uint8_t* dataout, uint8_t* datain, uint8_t len) override
+    {
+        spi_transfer(dataout, datain, len);
+    }
+
+    void SpiRead(uint8_t* datain, uint8_t len) override
+    {
+        spi_read(datain, len);
+    }
+
+    void SpiWrite(uint8_t* dataout, uint8_t len) override
+    {
+        spi_write(dataout, len);
+    }
 #endif
 
+ //-- RF power interface
+
+    void RfPowerCalc(int8_t power_dbm, uint8_t* sx_power, int8_t* actual_power_dbm) override
+    {
+#if defined DEVICE_HAS_I2C_DAC || defined DEVICE_HAS_INTERNAL_DAC_TWOCHANNELS
+        rfpower_calc(power_dbm, sx_power, actual_power_dbm, &dac);
+#else
+        sx1276_rfpower_calc(power_dbm, sx_power, actual_power_dbm, POWER_GAIN_DBM, POWER_SX1276_MAX_DBM);
+#endif
+    }
+
+    //-- init API functions
+
+    void _reset(void)
+    {
+        gpio_low(SX2_RESET);
+        delay_ms(5); // datasheet says > 100 us
+        gpio_high(SX2_RESET);
+        delay_ms(50); // datasheet says 5 ms
+    }
+
+    void Init(void)
+    {
+        Sx127xDriverCommon::Init();
+
+#ifndef DEVICE_HAS_DIVERSITY_SINGLE_SPI
+        spib_init();
+        spib_setnop(0x00); // 0x00 = NOP
+#else
+        // spi init done already by driver1
+#endif
+        sx2_init_gpio();
+        sx2_dio_exti_isr_clearflag();
+        sx2_dio_init_exti_isroff();
+
+        // no idea how long the SX1276 takes to boot up, so give it some good time
+        delay_ms(300);
+        _reset(); // this is super crucial ! was so for SX1280, is it also for the SX1276 ??
+
+        // this is not nice, figure out where to place
+#if defined DEVICE_HAS_I2C_DAC || defined DEVICE_HAS_INTERNAL_DAC_TWOCHANNELS
+        dac.Init();
+#endif
+
+        SetStandby(); // should be in STDBY after reset
+        delay_us(1000); // is this needed ????
+    }
+
+    //-- high level API functions
+
+    void StartUp(tSxGlobalConfig* global_config)
+    {
+//XX        // this is not nice, figure out where to place
```

Comment:
you can delete these comments here, it's sufficient to have them in teh driver above :)

Reply:
Removed comments from StartUp function